### PR TITLE
Add byo pki cosign support to docker image builds

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -237,49 +237,49 @@ commands:
...
-    description: "Add bash"
+  
+  do_docker_prep:
+    description: "Install packages and set commit."
     steps:
       - run:
-          name: install bash
+          name: install packages
+          command: |
+            apk add python3 py3-pip git cosign=2.0.1-r2 bash
```
Comment (invisig0th) [inline .circleci/config.yml:247]: do we need to add a ticket to un-pin this once they release?

File: synapse/tools/docker_validate.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import os
```
Comment (invisig0th) [inline synapse/tools/docker_validate.py:1]: Would we be better off making synapse/tools/docker/validate.py ? We can discuss :+1:

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/tools/docker_validate.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/tools/docker_validate.py:1]: That would be a fine change, especially if we ever offer any other tools

File: .circleci/config.yml (hunk 1 above)
Comment (vEpiphyte) [reply .circleci/config.yml:247]: No - that is the alpine 2.0.1 release identifier.
We could move it to `cosign~2.` to indicate any 2.x.x release.

File: docs/synapse/devguides/docker.rst [hunk 3]
```diff
@@ -97,4 +97,40 @@ Developers working with Synapse images should consider the following items:
 
     python -m pip install --break-system-packages yourTargetPackage
 
+Verifying image signatures
+--------------------------
+
+Synapse docker images which are release tagged ( e.g. ``:v2.1.3`` or
```
[the remark this answers is not in the document]
Comment (invisig0th) [reply docs/synapse/devguides/docker.rst:103]: Once this is fixed, it looks good to me!

[the remark this answers is not in the document]
Comment (vEpiphyte) [reply docs/synapse/devguides/docker.rst]: @mikemoritz Fixed

**Review by mikemoritz — APPROVED** [review]

**Pull request merged**